import sys
from PyQt5 import QtGui, QtWidgets, QtCore, uic
import os
import tempfile
class Ui_MainWindow(QtWidgets.QMainWindow):
    def setupUi(self, MainWindow):
        super().__init__()
        uic.loadUi('mainwindow.ui', self)
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(230, 300)
        MainWindow.setMinimumSize(QtCore.QSize(230, 300))
        MainWindow.setMaximumSize(QtCore.QSize(230, 300))
        MainWindow.setBaseSize(QtCore.QSize(2000, 2000))
        MainWindow.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.tabWidget = QtWidgets.QTabWidget(self.centralwidget)
        self.tabWidget.setGeometry(QtCore.QRect(0, 0, 400, 400))
        self.tabWidget.setMinimumSize(QtCore.QSize(400, 400))
        self.tabWidget.setMaximumSize(QtCore.QSize(800, 800))
        self.tabWidget.setBaseSize(QtCore.QSize(400, 400))
        self.tabWidget.setObjectName("tabWidget")
        self.tab = QtWidgets.QWidget()
        self.tab.setObjectName("tab")
        self.gridLayoutWidget = QtWidgets.QWidget(self.tab)
        self.gridLayoutWidget.setGeometry(QtCore.QRect(0, 0, 231, 241))
        self.gridLayoutWidget.setObjectName("gridLayoutWidget")
        self.gridLayout = QtWidgets.QGridLayout(self.gridLayoutWidget)
        self.gridLayout.setContentsMargins(0, 0, 0, 0)
        self.gridLayout.setObjectName("gridLayout")
        self.verticalLayout_4 = QtWidgets.QVBoxLayout()
        self.verticalLayout_4.setObjectName("verticalLayout_4")
        spacerItem = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout_4.addItem(spacerItem)
        self.vichisl = QtWidgets.QPushButton(self.gridLayoutWidget)
        self.vichisl.setObjectName("vichisl")
        self.verticalLayout_4.addWidget(self.vichisl)
        self.ochist = QtWidgets.QPushButton(self.gridLayoutWidget)
        self.ochist.setObjectName("ochist")
        self.verticalLayout_4.addWidget(self.ochist)
        self.sohran = QtWidgets.QPushButton(self.gridLayoutWidget)
        self.sohran.setObjectName("sohran")
        self.verticalLayout_4.addWidget(self.sohran)
        spacerItem1 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout_4.addItem(spacerItem1)
        self.gridLayout.addLayout(self.verticalLayout_4, 1, 1, 1, 1)
        self.verticalLayout_2 = QtWidgets.QVBoxLayout()
        self.verticalLayout_2.setSpacing(0)
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        spacerItem2 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout_2.addItem(spacerItem2)
        self.label = QtWidgets.QLabel(self.gridLayoutWidget)
        self.label.setObjectName("label")
        self.verticalLayout_2.addWidget(self.label)
        self.get_a = QtWidgets.QLineEdit(self.gridLayoutWidget)
        self.get_a.setContextMenuPolicy(QtCore.Qt.PreventContextMenu)
        self.get_a.setText("")
        self.get_a.setMaxLength(8)
        self.get_a.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.get_a.setObjectName("get_a")
        self.verticalLayout_2.addWidget(self.get_a)
        spacerItem3 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.verticalLayout_2.addItem(spacerItem3)
        self.line = QtWidgets.QFrame(self.gridLayoutWidget)
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.verticalLayout_2.addWidget(self.line)
        self.label_2 = QtWidgets.QLabel(self.gridLayoutWidget)
        self.label_2.setObjectName("label_2")
        self.verticalLayout_2.addWidget(self.label_2)
        self.get_b = QtWidgets.QLineEdit(self.gridLayoutWidget)
        self.get_b.setContextMenuPolicy(QtCore.Qt.PreventContextMenu)
        self.get_b.setText("")
        self.get_b.setMaxLength(8)
        self.get_b.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.get_b.setObjectName("get_b")
        self.verticalLayout_2.addWidget(self.get_b)
        spacerItem4 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout_2.addItem(spacerItem4)
        self.gridLayout.addLayout(self.verticalLayout_2, 0, 0, 1, 1)
        self.verticalLayout_3 = QtWidgets.QVBoxLayout()
        self.verticalLayout_3.setObjectName("verticalLayout_3")
        spacerItem5 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout_3.addItem(spacerItem5)
        self.label_3 = QtWidgets.QLabel(self.gridLayoutWidget)
        self.label_3.setObjectName("label_3")
        self.verticalLayout_3.addWidget(self.label_3)
        self.radioButton_sloj = QtWidgets.QRadioButton(self.gridLayoutWidget)
        self.radioButton_sloj.setObjectName("radioButton_sloj")
        self.verticalLayout_3.addWidget(self.radioButton_sloj)
        self.radioButton_vich = QtWidgets.QRadioButton(self.gridLayoutWidget)
        self.radioButton_vich.setObjectName("radioButton_vich")
        self.verticalLayout_3.addWidget(self.radioButton_vich)
        self.radioButton_ymnoj = QtWidgets.QRadioButton(self.gridLayoutWidget)
        self.radioButton_ymnoj.setObjectName("radioButton_ymnoj")
        self.verticalLayout_3.addWidget(self.radioButton_ymnoj)
        self.radioButton_del = QtWidgets.QRadioButton(self.gridLayoutWidget)
        self.radioButton_del.setObjectName("radioButton_del")
        self.verticalLayout_3.addWidget(self.radioButton_del)
        spacerItem6 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout_3.addItem(spacerItem6)
        self.gridLayout.addLayout(self.verticalLayout_3, 0, 1, 1, 1)
        self.tabWidget.addTab(self.tab, "")
        self.tab_2 = QtWidgets.QWidget()
        self.tab_2.setObjectName("tab_2")
        self.show_history_vich = QtWidgets.QTextEdit(self.tab_2)
        self.show_history_vich.setGeometry(QtCore.QRect(0, 0, 241, 231))
        self.show_history_vich.setReadOnly(True)
        self.show_history_vich.setObjectName("show_history_vich")
        self.tabWidget.addTab(self.tab_2, "")
        MainWindow.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 230, 18))
        self.menubar.setObjectName("menubar")
        self.menu = QtWidgets.QMenu(self.menubar)
        self.menu.setObjectName("menu")
        self.menu_2 = QtWidgets.QMenu(self.menubar)
        self.menu_2.setObjectName("menu_2")
        self.menu_3 = QtWidgets.QMenu(self.menubar)
        self.menu_3.setObjectName("menu_3")
        MainWindow.setMenuBar(self.menubar)
        self.Open = QtWidgets.QAction(MainWindow)
        self.Open.setObjectName("Open")
        self.Exit = QtWidgets.QAction(MainWindow)
        self.Exit.setObjectName("Exit")
        self.about_program = QtWidgets.QAction(MainWindow)
        self.about_program.setObjectName("about_program")
        self.about_author = QtWidgets.QAction(MainWindow)
        self.about_author.setObjectName("about_author")
        self.actionsfdg = QtWidgets.QAction(MainWindow)
        self.actionsfdg.setObjectName("actionsfdg")
        self.Styles = QtWidgets.QAction(MainWindow)
        self.Styles.setObjectName("Styles")
        self.menu.addAction(self.Open)
        self.menu.addSeparator()
        self.menu.addAction(self.Exit)
        self.menu_2.addAction(self.about_program)
        self.menu_2.addAction(self.about_author)
        self.menu_3.addAction(self.Styles)
        self.menubar.addAction(self.menu.menuAction())
        self.menubar.addAction(self.menu_3.menuAction())
        self.menubar.addAction(self.menu_2.menuAction())

        self.retranslateUi(MainWindow)
        self.tabWidget.setCurrentIndex(0)
        self.ochist.pressed.connect(self.get_a.clear) # type: ignore
        self.ochist.pressed.connect(self.get_b.clear) # type: ignore
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "КалькуляторQtDesigner"))
        self.vichisl.setText(_translate("MainWindow", "Вычислить"))
        self.ochist.setText(_translate("MainWindow", "Очистить"))
        self.sohran.setText(_translate("MainWindow", "Сохранить"))
        self.label.setText(_translate("MainWindow", "Коэффициент а"))
        self.get_a.setPlaceholderText(_translate("MainWindow", "Коэффициент а"))
        self.label_2.setText(_translate("MainWindow", "Коэффициент b"))
        self.get_b.setPlaceholderText(_translate("MainWindow", "Коэффициент b"))
        self.label_3.setText(_translate("MainWindow", "Вид выражения"))
        self.radioButton_sloj.setText(_translate("MainWindow", "Сложение"))
        self.radioButton_vich.setText(_translate("MainWindow", "Вычитание"))
        self.radioButton_ymnoj.setText(_translate("MainWindow", "Умножение"))
        self.radioButton_del.setText(_translate("MainWindow", "Деление"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab), _translate("MainWindow", "Решение"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab_2), _translate("MainWindow", "История"))
        self.menu.setTitle(_translate("MainWindow", "Файл"))
        self.menu_2.setTitle(_translate("MainWindow", "Справка"))
        self.menu_3.setTitle(_translate("MainWindow", "Стиль вывода"))
        self.Open.setText(_translate("MainWindow", "Открыть"))
        self.Exit.setText(_translate("MainWindow", "Выход"))
        self.about_program.setText(_translate("MainWindow", "О программе"))
        self.about_author.setText(_translate("MainWindow", "Об авторе"))
        self.actionsfdg.setText(_translate("MainWindow", "sfdg"))
        self.Styles.setText(_translate("MainWindow", "Открыть"))
def main():
    app = QtWidgets.QApplication(sys.argv)
    window = Ui_MainWindow()
    #window.__init__()
    window.show()
    app.exec()


if __name__ == '__main__':
    main()